### PR TITLE
Downgrade husky to major version 4

### PR DESCRIPTION
**Pull request by mrcnski**
As discussed on Discord `husky` uses a Open Parity license that is incompatible with our project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
   "lint-staged": {
     "*.{js,jsx,ts,tsx}": [
       "eslint --ext .ts --fix",
-      "tsc --esModuleInterop --noemit",
+      "tsc --esModuleInterop",
```
Comment (interstella-5555) [inline package.json:45]: should this be included in the downgrade? is this interconnected?

Comment (mrcnski) [reply package.json:45]: No harm in having it here, it's a minor change. tsconfig.json already has noemit. I can also just remove this change, doesn't matter either way.

**Review by interstella-5555 — DISMISSED** [review]

**Comment by mrcnski** [comment]
I have just downgraded `url-parse` as well. Update `1.4.7 -> 1.5.0` was released 21 hours ago and it broke our tests. I don't have time to go fix all these tests just because some dependency doesn't respect semver so I set it to a fixed version.

File: package.json [hunk 2]
```diff
@@ -92,15 +92,15 @@
...
+    "@types/url-parse": "=1.4.3",
     "@typescript-eslint/eslint-plugin": "^4.3.0",
     "@typescript-eslint/parser": "^4.3.0",
     "axios-mock-adapter": "^1.18.2",
     "babel-plugin-transform-class-properties": "^6.24.1",
     "eslint": "^7.11.0",
     "eslint-plugin-compat": "^3.8.0",
     "eslint-plugin-jsdoc": "^32.0.0",
-    "husky": "^4.3.8",
+    "husky": "=4.3.8",
```
Comment (interstella-5555) [inline package.json]:
```suggestion
    "husky": "4.3.8",
```

File: package.json [hunk 3]
```diff
@@ -92,15 +92,15 @@
     "@types/randombytes": "^2.0.0",
     "@types/sjcl": "^1.0.29",
     "@types/url-join": "^4.0.0",
-    "@types/url-parse": "^1.4.3",
+    "@types/url-parse": "=1.4.3",
```
Comment (interstella-5555) [inline package.json]:
```suggestion
    "@types/url-parse": "1.4.3",
```

File: package.json [hunk 4]
```diff
@@ -76,7 +76,7 @@
     "sjcl": "^1.0.8",
     "tweetnacl": "^1.0.3",
     "url-join": "^4.0.1",
-    "url-parse": "^1.4.7"
+    "url-parse": "=1.4.7"
```
Comment (interstella-5555) [inline package.json]:
```suggestion
    "url-parse": "1.4.7"
```

**Review by interstella-5555 — CHANGES_REQUESTED** [review]
You don't need to use `=` in package.json to bind library to specific version, it's not really a common practice to do so too, haven't seen anyone do that actually. Otherwise it's 👍

**Comment by interstella-5555** [comment]
You will need to rerun `yarn` to update the lock file too @m-cat

**Comment by ro-tex** [comment]
Can we add a note at a visible place that explains that we need to stay at version 4 because of licensing? Otherwise we might accidentally upgrade in the future. Same for `url-parse`.

Maybe a dedicated `DEPENDENCIES.md`?

**Review by ro-tex — APPROVED** [review]

**Comment by dghelm** [comment]
Will `skynet-js` need to use the same license as `skyd` or can we just keep it at MIT?

Edit: Confirmed with David that `skynet-js` can continue to be MIT, it's just `skyd` that will be Skynet Licensed

**Comment by mrcnski** [comment]
Closing since this repo is staying on MIT, which should be fine with Parity License. Can reopen if needed.

**Pull request closed without merging**